packages/shared/src/components/comments/CommentBox.tsx

```diff
@@ -76,11 +79,38 @@ function CommentBox({
         <div className="flex flex-col ml-3 typo-callout">
           <FlexRow>
             <CommentAuthor
-              postAuthorId={postAuthorId}
               author={comment.author}
               appendTooltipTo={appendTooltipTo}
+              badges={[
+                <SquadMemberRoleBadge
+                  key="squadMemberRole"
+                  source={post.source}
+                  author={comment.author}
+                />,
+                comment.author.id === postAuthorId && (
+                  <UserBadge
+                    key="author"
+                    className="text-theme-status-help"
+                    content="Author"
+                    Icon={FeatherIcon}
+                    iconProps={{
+                      secondary: true,
+                    }}
+                  />
+                ),
+                comment.author.id === postScoutId && (
+                  <UserBadge
+                    key="scout"
+                    className="text-theme-color-bun"
+                    content="Scout"
+                    Icon={ScoutIcon}
+                    iconProps={{
+                      secondary: true,
+                    }}
+                  />
+                ),
+              ].filter(Boolean)}
```

Comment:
Because in the first iteration I was using functions here and rendering them inside CommentBox this was just to avoid having false values and need to do type checking inside CommentBox. For this `false` is just fine as it does not render by React. good catch 👌